provider/aws/aws.go

```diff
@@ -372,14 +543,67 @@ func (p *AwsProvider) createCFStack(nets []string, spec *stackSpec) (string, err
 					err = provider.NewAlreadyExistsError(fmt.Sprintf("%s AlreadyExists", stackName))
 				}
 			}
-			return spec.name, err
+			return err
 		}
-		return aws.StringValue(resp.StackId), nil
+		ctx, cancel := context.WithTimeout(context.Background(), maxStackWaitTimeout)
+		defer cancel()
+		return p.waitForStack(ctx, stackStatusCheckInterval, stackName)
 	}
-	log.Debugf("%s: DRY: Stack to create: %s", p, params)
-	log.Debugln(aws.StringValue(params.TemplateBody))
-	return "DRY stackID", nil
+	p.logger.Debugf("%s: DRY: Stack to create: %s", p, params)
+	p.logger.Debugln(aws.StringValue(params.TemplateBody))
+	return nil
+
+}
 
+func (p *AWSProvider) getStackByName(stackName string) (*cloudformation.Stack, error) {
+	params := &cloudformation.DescribeStacksInput{
+		StackName: aws.String(stackName),
+	}
+	resp, err := p.cloudformation.DescribeStacks(params)
+	if err != nil {
+		return nil, err
+	}
+	//we expect only one stack
+	if len(resp.Stacks) != 1 {
+		return nil, fmt.Errorf("unexpected response, got %d, expected 1 stack", len(resp.Stacks))
+	}
+	return resp.Stacks[0], nil
+}
+
+func (p *AWSProvider) waitForStack(ctx context.Context, waitTime time.Duration, stackName string) error {
+	for {
+		stack, err := p.getStackByName(stackName)
+		if err != nil {
+			return err
+		}
+		switch *stack.StackStatus {
+		case cloudformation.StackStatusUpdateComplete:
+			return nil
+		case cloudformation.StackStatusCreateComplete:
+			return nil
+		case cloudformation.StackStatusDeleteComplete:
+			return nil
+		case cloudformation.StackStatusCreateFailed:
+			return errCreateFailed
+		case cloudformation.StackStatusDeleteFailed:
+			return errDeleteFailed
+		case cloudformation.StackStatusRollbackComplete:
+			return errRollbackComplete
+		case cloudformation.StackStatusRollbackFailed:
+			return errRollbackFailed
+		case cloudformation.StackStatusUpdateRollbackComplete:
+			return errUpdateRollbackComplete
+		case cloudformation.StackStatusUpdateRollbackFailed:
+			return errUpdateRollbackFailed
+		}
+		p.logger.Debugf("Stack '%s' - [%s]", stackName, *stack.StackStatus)
```

Comment:
maybe nice to have stringer interface implemented for StackStatus

Reply:
The Stack struct is from the aws Go SDK, so we can't really do that. I will use `aws.StringValue` instead which is at least a bit more clean.